app/code/community/Oyst/OneClick/controllers/NotificationsController.php

```diff
@@ -48,30 +48,56 @@ public function indexAction()
             return $this->badRequest('No event or data');
         }
 
-        switch ($event) {
-            case 'order.v2.new':
-                $modelName = 'oyst_oneclick/order';
-                break;
-            case 'order.shipments.get':
-                $modelName = 'oyst_oneclick/catalog';
-                break;
-            default:
-                return $this->badRequest('Event name ' . $event . ' is not allow.');
-                break;
-        }
+        if ($event == 'order.v2.new' || $event == 'order.shipments.get') {
```

Comment:
Please use only switch/case to make a notification routing.
Add ```notification.newOrder``` and ```payment``` as new case, and move all helper logic into a model.